### ISSUE TITLE
Tags datagen crashes when referencing forge tags

### ISSUE THREAD

**Issue by sciwhiz12** [MinecraftForge/MinecraftForge#7320]
**Minecraft Version:** 1.16.3 _(issue may also occur on 1.16.2)_

**Forge Version:** 34.0.0 _(as stated, may also occur with 33.0.61)_

**Logs:** [debug.log, in mod development environment][crash_log]

**Steps to Reproduce:**
 1. Clone [this branch][broken_datagen_branch] of my mod repository
_An alternative to this step is to extend `BlockTagsProvider`, and create a custom tag that references one of the Forge-added tags **(not vanilla ones, they work fine)**_
 2. Setup the mod as normal (import, run `gen***Runs`)
 3. Run the `runData` task
 4. _Observe that the run fails with an `IllegalArgumentException: Couldn't define tag ... as it is missing following references: ...`_

**Description of issue:**
_Follow up from [this Discord conversation on the Forge Project discord][discord_convo]_

When referencing a Forge-added tag in a tag builder during data generation, the tags data provider will fail to find the existing tags on disk (inside the Forge jar), and will crash. 
Before I updated to 1.16.3 and tried the new method provided in #7271, my 1.16.x branch extended `ForgeBlockTagsProvider` and overrode `makePath` to ensure only my tag files were written to disk (this previous solution did work as expected).

Here is the exception from my debug.log: 
```
java.lang.IllegalArgumentException: Couldn't define tag voxeltools:ground as it is missing following references: #forge:cobblestone (from voxeltools),#forge:dirt (from voxeltools),#forge:end_stones (from voxeltools),#forge:ores (from voxeltools),#forge:stone (from voxeltools),#forge:sandstone (from voxeltools)
```

[crash_log]: https://gist.github.com/sciwhiz12/723a63a0ed61f0a2084dad4953a922b8
[broken_datagen_branch]: https://github.com/sciwhiz12/VoxelTools/tree/broken_tags_datagen
[discord_convo]: https://discordapp.com/channels/313125603924639766/725850371834118214/753949471422808075

**Comment by ChampionAsh5357** [comment]
Just to add a little extension, it seems that minecraft is the only hardcoded resource into `ExistingFileHelper`. All other resources must be passed in through `build.gradle` including forge assets/datapacks. Since none of this is passed in, any addition of the `ExsitingFileHelper` will not allow resources to be taken in from dependent mods. You would need the ability to parse the assets/datapacks from each jar which doesn't bode well for any updates in mappings or versions.

**Comment by sciwhiz12** [comment]
Tested the same branch with an update to `1.16.4, 35.1.7, 20201028-1.16.3` (along with edits to code for now-mapped names), and the run now succeeds as expected. Closing as fixed by #7456.
